docs/proposals/control-plane-config.md

```diff
@@ -17,3 +17,86 @@ option that we will support is log level.
 ## Non-Goals
 
 This proposal is *not* defining a way to dynamically configure the data plane.
+
+## Introduction
+
+The NKG control plane will evolve to have various user-configurable options. These could include, but are not
+limited to, log level, tracing, or metrics. For the best user experience, these options should be able to be
+changed at runtime, to avoid having to restart NKG. The first option that we will allow users to configure is the
+log level. The easiest and most intuitive way to implement a Kubernetes-native API is through a CRD.
+
+## API, Customer Driven Interfaces, and User Experience
+
+The API would be provided in a CRD. An authorized user would interact with this CRD using `kubectl` to `get`
+or `edit` the configuration.
+
+Proposed configuration CRD example:
+
+```yaml
+apiVersion: nginx.gateway.k8s.io/v1beta1
+kind: NGINXControlConfig
+metadata:
+    name: nkg-config
+    namespace: nginx-gateway
+spec:
+    logLevel: info
+    ...
+```
+
+- The CRD would be Namespace-scoped, living in the same Namespace as the controller that it applies to.
+- CRD is initialized and created when NKG is deployed
+- NKG references the name of this CRD via CLI arg, and only watches this CRD
+- If user deletes resource, NKG logs an error and creates an event. Last state is used until CRD is restored.
+
+For discussion with team:
```

Comment:
Yep, I forgot to add that to the yaml

Reply:
Do we think status is necessary at this point? Is it possible to fail to change the log level?

I know the api is v1alpha1, but I think we should avoid adding fields unless we have an explicit use case for it.

Reply:
Probably not necessary right now, but I think status is usually expected on CRDs (though not required), even if it's always a "Good" status.

And in my initial implementation, there could be an error returned when parsing the log level (though our openapi would prevent it). But it technically still exists.

Reply:
If we include it now, what would it look like?

Reply:
I'd include it because status also tells the user if NKG handled that resource even if there was no error.

Although unlikely, validation can still fail, because OpenAPI scheme can be changed for our CRD.

Reply:
I think for now it would just contain the Conditions array.